Assignment1.1.SQL

```diff
@@ -0,0 +1,238 @@
+
+MUSA 508 
+Olivia Scalora
+Assignment 1
+09/19/21
+
+/*
+ 1. Example: How many bike trips in Q2 2019
+*/
+
+SELECT COUNT(*)
+    FROM indego_trips_2019_q2
+
+--> 	
+206354 bike trips
+
+/*
+  2. What is the percent change in trips in Q2 2020 as compared to Q2 2019?
+  Using only the table from Q2 2020 and the number calculated in the previous
+  question, find the percent change of number of trips in Q2 2020 as compared
+  to 2019. Remember you can do calculations in the select clause.
+*/
+
+WITH count2019 AS (
+    SELECT 0 AS id,
+    COUNT(*) AS trips19
+    FROM indego_trips_2019_q2
+    ) ,
+
+    count2020 AS (
+    SELECT 0 AS id, 
+    COUNT(*) AS trips20
+    FROM indego_trips_2020_q2
+    )
+
+SELECT count2019, 
+       count2020, 
+       ((trips20::FLOAT-trips19::FLOAT)/trips19*100)::TEXT || '%'
+            AS pctchange
+    FROM count2019
+    INNER JOIN count2020
+    ON count2019.id=count2020.id
+
+
+/*
+  Bonus: If you want to get fancier here, you can cast the result to a string
+  and concatenate a '%' to the end. For example:
+      (10 + 3.2)::text || '%' AS perc_change
+  This uses the type casting (number to string) and string concatenation
+  operator (`||`, double pipes) that's essentially a `+` for strings.
+*/
+
+/*
+ 3. What is the average duration of a trip for 2019?
+*/
+
+SELECT AVG(duration)
+    FROM indego_trips_2019_q2
+
+  
+/*
+ 4. What is the average duration of a trip for 2020?
+*/
+
+SELECT AVG(duration)
+    FROM indego_trips_2020_q2
+
+
+/*
+  5. What is the longest duration trip?
+*/
+
+WITH trip_duration AS (
+    SELECT trip_id, duration, start_time::date
+    FROM indego_trips_2019_q2
+    UNION All
+    SELECT trip_id, duration, start_time::date
+    FROM indego_trips_2020_q2)
+
+SELECT MAX(duration)
+    FROM trip_duration
+
+
+/*
+ 6. How many trips were shorter than 10 minutes?
+*/
+
+WITH trip_duration_u_19_20 AS (
+    SELECT trip_id, duration
+    FROM indego_trips_2019_q2
+    UNION All
+    SELECT trip_id, duration
+    FROM indego_trips_2020_q2)
+
+SELECT COUNT(duration)
+    FROM trip_duration_u_19_20
+    WHERE duration < 10
+
+
+/*
+ 7. How many trips started on one day and ended in the next?
+*/
+
+WITH trip_start_time_u_19_20 AS (
+    SELECT trip_id, 
+        DATE_PART('day',start_time::date) AS start_day, 
+        DATE_PART('day',end_time::date) AS end_day
+    FROM indego_trips_2019_q2
+    UNION ALL
+    SELECT trip_id,
+        DATE_PART('day',start_time::date) AS start_day, 
+        DATE_PART('day',end_time::date) AS end_day 
+    FROM indego_trips_2020_q2)
+
+SELECT COUNT(*) 
+    FROM trip_start_time_u_19_20
+    WHERE start_day != end_day
```

Comment:
Right track, but slightly over-counting. There are actually trips in the data where the start end end dates are more than 1 day apart. The solution I used is [here](https://github.com/musa-509-fall-2021/assignment-01-intro-to-geographic-queries/blob/answers/query07.sql#L16-L25) (or a more compact equivalent solution [here](https://github.com/musa-509-fall-2021/assignment-01-intro-to-geographic-queries/blob/answers/query07.sql#L40-L42))